src/admin/aws/upload.integration.ts

```diff
@@ -25,12 +37,21 @@ describe('Upload', () => {
 
     const upload = await uploadImageToS3(s3, image);
 
-    // Check that the returned url matches the expected pattern
-    // http://localstack:4566/curated-corpus-api-local-images/some-random-path.jpeg
-    const urlPrefix = config.aws.s3.localEndpoint;
-    const urlPattern = new RegExp(
-      `^${urlPrefix}/${config.aws.s3.bucket}/.+.jpeg$`
+    expectSuccessfulUpload(upload);
+  });
+
+  it('uploads an image to s3 using the url', async () => {
+    nock('https://image.com')
+      .get('/dancing-in-the-air.jpeg')
+      .replyWithFile(200, testFilePath, {
+        'Content-Type': 'image/jpeg',
+      });
```

Comment:
I like this! Just clarify, we are mocking the `fetch()` call here right?

Reply:
we are mocking the HTTP request, yes